transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringEventLoop.java

```diff
@@ -120,10 +120,31 @@ protected void run() {
 
         // Lets add the eventfd related events before starting to do any real work.
         submissionQueue.addPollIn(eventfd.intValue());
-        submissionQueue.submit();
 
         for (;;) {
             logger.trace("Run IOUringEventLoop {}", this.toString());
+            // avoid blocking for as long as possible - loop until two consecutive "empty" results
+            boolean maybeMoreTasks = true;
+            boolean maybeMoreIo;
+            do {
+                maybeMoreIo = completionQueue.process(this) != 0;
+                if (!maybeMoreIo & !maybeMoreTasks) {
+                    break; // no need to check tasks again here
+                }
+                maybeMoreTasks = runAllTasks();
+                try {
+                    if (isShuttingDown()) {
+                        closeAll();
+                        if (confirmShutdown()) {
+                            return;
+                        }
+                    }
+                } catch (Throwable t) {
+                    logger.info("Exception error: {}", t);
+                }
+            } while (maybeMoreIo | maybeMoreTasks);
```

Comment:
```suggestion
            boolean maybeMoreIo = completionQueue.process(this) != 0;
            while (maybeMoreIo || maybeMoreTasks) {
                maybeMoreTasks = runAllTasks();
                try {
                    if (isShuttingDown()) {
                        closeAll();
                        if (confirmShutdown()) {
                            return;
                        }
                    }
                } catch (Throwable t) {
                    logger.info("Exception error: {}", t);
                }
                maybeMoreIo = completionQueue.process(this) != 0;
            }
```

Reply:
@nitsanw I like your suggested change :+1: I think it's cleaner to avoid break's

Reply:
Yeah, the do-while only made sense when we had the simple loop previously.

Reply:
I was aiming to allow breaking out after either the tasks run or the io run, since there could be a redundant attempt otherwise (for example some tasks run, then CQ is empty, then no tasks are run - the proposed version will attempt to process the CQ again unnecessarily). In practice I guess it wouldn't make noticeable difference, so maybe fine to stick with slightly simpler logic?

Reply:
let's follow "KISS" rule here and keep it simple